testsuite/do_tests.sh

```diff
@@ -408,6 +408,10 @@ if test "${MUSIC}"; then
         sh_file="${TESTDIR}/$(basename ${music_file} .music).sh"
         if test ! -f "${sh_file}"; then sh_file=""; fi
 
+        # Check if there is an accompanying input data file
+        input_file="${TESTDIR}/$(basename ${music_file} .music)0.dat"
```

Comment:
I don't get the logic here. Why the 0 at the end? Would it not be better to grep the music file to get the name of the input file from this line
```
# input file: test_message0.dat
```

Reply:
That's what I did initially. But if there is no data file specified, grep fails with a nonzero exit code and installcheck crashes. Instead  of jumping through hoops to avid the exit code, we simply assume here that there is a datafile and unset `input_file` if there isn't.

The 0 has to do with how MUSIC somehow handles input files:
https://github.com/nest/nest-simulator/blob/a06b43c9ea67973a17774af21c9c4f5f08a4ecdb/testsuite/musictests/test_message.music#L26-L31